netXMux.go

```diff
@@ -306,14 +311,22 @@ func (mux *XMux) HandleFunc(uriHandler *UriHandler) {
 }
 
 // RemoveHandleFunc removes the handler functions for the given pattern.
-func (mux *XMux) RemoveHandleFunc(uriPattern string) {
+func (mux *XMux) RemoveHandleFunc(method string, uriPattern string) {
 	mux.mu.Lock()
 	defer mux.mu.Unlock()
 
-	delete(mux.m, uriPattern)
+	key := makeKey(method, uriPattern)
+	keyNoTrailingSlash := makeKey(method, uriPattern[0:len(uriPattern)-1])
```

Comment:
probably worth validating against an empty uriPattern and logging an error instead of crashing here?

Reply:
Good point! Hopefully you feel it's addressed in 542b2f849d84cd869e52f382d0e2dd80135b28f2.